cms/djangoapps/contentstore/helpers.py

```diff
@@ -375,10 +385,21 @@ def _import_xml_node_to_parent(
     if copied_from_block:
         # Store a reference to where this block was copied from, in the 'copied_from_block' field (AuthoringMixin)
         temp_xblock.copied_from_block = copied_from_block
+    if copied_from_block and link_to_upstream:
+        # If requested, link this block as a downstream of where it was copied from
+        temp_xblock.upstream = copied_from_block
+        try:
+            sync_from_upstream(temp_xblock, user, apply_updates=False)
```

Comment:
I would appreciate a comment above this `sync_from_upstream` line that says what it's supposed to do here. Because it's not actually syncing, right? It's (I think) mostly setting the important `upstream_version` field, but also storing the original values of the upstream fields ?

I think we also need to call out a common case that needs to be handled carefully: if the pasted block is from a draft version of the block, we have actually pasted a newer version than is currently available upstream. In that case, we don't want `downstream.upstream_version = upstream_link.version_available` but rather `downstream.upstream_version = copied_from_block.draft_version` or something like that.

Reply:
@bradenmacdonald Thanks for calling this out. I was trying to treat this as a special case of `sync_from_upstream`, but that's really not right--it's not a sync at all. All we want to do after pasting is:
* Set the upstream to the copied block's usage key
* Set the upstream_version to the copied block's version. I added a `version_num` field to `StagedContentData` to handle this (it'll be 0 for things without versions, like course blocks).
* Fetch the upstream-set values for customizable fields, and save them into the pasted block's `upstream_*`fields. I've factored out a new function for this, although I'm thrashing on the name a bit. I think I'll call it `fetch_upstream_values_for_customizable_fields(...)` unless you have any objections.